wireapp/wire-ios-mono
# fix: If clipboard is not enabled, do not show copy action for media links - WPB-4982

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4982" title="WPB-4982" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-4982</a>  iOS Column 1 Build - Link preview is shown on long press
  </td></table>
  <br />
 























----
#### PR Submission Checklist for internal contributors

- [Fix] If clipboard is not enabled, do not show copy action for media links for C1 builds - WPB-3250

- The **PR Description**
  - On long press on C1 builds, copying of links to System clipboard should not be supported. In this PR, we check if the Clipboard is enabled/disabled. Based on which we show the context menu.
----

# What's new in this PR?

### Issues

- Users are able to copy links to the System Clipboard

### Causes (Optional)

_Earlier we used to show system context menu which could leak the links to be copied to the system Clipboard._

### Solutions

_We create a custom context menu based on the Settings for Clipboard._

### Testing

#### Test Coverage (Optional)


#### How to Test

1. Change the Settings file to disable Clipboard and run the app.
2. Go to a group or a contact
3. Enter a link to the group or the person conversation
4. Long press on the link
5. Context menu with options for Visit Link, Reply, Details, Edit and Cancel should be displayed

### Attachments (Optional)


|Before|After|
|--|--|
|![Screenshot 2023-10-06 at 09 51 03](https://github.com/wireapp/wire-ios/assets/17357739/762b6d45-eb58-47e4-b482-df122e406465)|![Simulator Screenshot - iPhone 15 - 2023-10-26 at 10 00 53](https://github.com/wireapp/wire-ios/assets/17357739/f3aad2de-ba06-4678-a9a7-5155d193dd8f)|


----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
